src/v1/reset_password.rs

```diff
@@ -0,0 +1,39 @@
+//! [POST /_synapse/admin/v1/reset_password/:user_id](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst#reset-password)
+
+use ruma::api::ruma_api;
+use ruma::identifiers::UserId;
+
+ruma_api! {
+    metadata: {
+        description: "password reset endpoint",
+        method: POST,
+        name: "reset_password",
+        path: "/_synapse/admin/v1/reset_password/:user_id",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    request: {
+        /// user ID of the account to reset the password
+        #[ruma_api(path)]
+        pub user_id: &'a UserId,
+
+        /// new password
+        pub new_password: String,
+
+        /// Logout all devices of the user, so it necessary to login with the new password again.
+        /// Defaults to true.
+        #[serde(default = "ruma::serde::default_true")]
+        pub logout_devices: bool,
+    }
+
+    response: {}
+
+}
+
+impl<'a> Request<'a> {
+    /// Creates an `Request` with the given user ID and the new password.
+    pub fn new(user_id: &'a UserId, new_password: String) -> Self {
+        Self { user_id, new_password, logout_devices: true }
```

Comment:
rustfmt complains about this but it's actually the style I would use too. I just imported ruma's `.rustfmt.toml` into this repository, after a rebase rustfmt should not complain anymore.